core/io/http_client.cpp

```diff
@@ -96,7 +96,12 @@ Error HTTPClient::request_raw(Method p_method, const String &p_url, const Vector
 	};
 
 	String request = String(_methods[p_method]) + " " + p_url + " HTTP/1.1\r\n";
-	request += "Host: " + conn_host + ":" + itos(conn_port) + "\r\n";
+	if (ssl && conn_port == 443) {
```

Comment:
I think you should also not add port 80 in case it is non-ssl.
Something like: (untested)
```c++
if ((ssl && conn_port == 443) || (!ssl && conn_port == 80)) {
	…
```
(applies to other if below as well)